community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/staging/DynamicProcessorAssigner.java

```diff
@@ -81,7 +76,7 @@ public void check( StageExecution[] executions )
                     permits -= assignProcessorsToPotentialBottleNeck( execution, permits );
                 }
                 // Be a little more conservative removing processors from too fast steps
-                if ( removeProcessorFromPotentialIdleStep( execution ) )
+                if ( permits == 0 && removeProcessorFromPotentialIdleStep( execution ) )
```

Comment:
I believe there were tests changed for this, tests that already set up such a scenario and which now behave differently.